packages/frontend-2/components/projects/Dashboard.vue

```diff
@@ -170,35 +165,19 @@ onUserProjectsUpdate((res) => {
 
   if (isNewProject && incomingProject) {
     // Add to User.projects where possible
-    modifyObjectFields<UserProjectsArgs, User['projects']>(
+    modifyObjectField(
       cache,
       getCacheId('User', activeUserId),
-      (fieldName, variables, value, { ref }) => {
-        if (fieldName !== 'projects') return
-        if (variables.filter?.search?.length) return
-        if (variables.filter?.onlyWithRoles?.length) {
-          const roles = variables.filter.onlyWithRoles
-          if (!roles.includes(incomingProject.role || '')) return
-        }
-
-        return {
-          ...value,
-          items: [ref('Project', incomingProject.id), ...(value.items || [])],
-          totalCount: (value.totalCount || 0) + 1
-        }
-      }
-    )
-
-    // Elsewhere - just evict fields directly
-    evictObjectFields<UserProjectsArgs, User['projects']>(
-      cache,
-      getCacheId('User', activeUserId),
-      (fieldName, variables) => {
-        if (fieldName !== 'projects') return false
-        if (variables.filter?.search?.length) return true
-
-        return false
-      }
+      'projects',
+      ({ helpers: { ref, createUpdatedValue } }) =>
+        createUpdatedValue(({ update }) => {
+          update('items', (items) => [
+            ref('Project', incomingProject.id),
+            ...(items || [])
+          ])
+          update('totalCount', (count) => count + 1)
+        }),
+      { autoEvictFiltered: true }
```

Comment:
Option that automatically evicts all fields that have variables w/ filters in them. You usually want to do this when adding a new item to a list - it may be hard to tell if the item fits the filtered query, and it'll just be easier to evict that version of the query and refetch it 

cc @cdriesler